python/src/pipeline/models.py

```diff
@@ -17,12 +17,48 @@
 """Datastore models used by the Google App Engine Pipeline API."""
 
 from google.appengine.ext import db
+from google.appengine.ext import ndb
 from google.appengine.ext import blobstore
 
 # Relative imports
 import simplejson
 import util
 
+def strip_accents(s):
+  """Strips unicode accents from a string"""
+  import unicodedata
+  if s is None:
+    return ''
+  if isinstance(s, str):
+    return s
+  return ''.join(c for c in unicodedata.normalize('NFKD', s)
+                 if unicodedata.category(c) != 'Mn')
+
+def truncate_value(value):
+  """Shorten the given value, preserving dict structure."""
+  if isinstance(value, dict):
+    return dict((k, truncate_value(v)) for k, v in value.iteritems())
+  elif isinstance(value, ndb.Key):
+    return value.urlsafe()
+  elif isinstance(value, db.Key):
+    return str(value.urlsafe)
+  else:
+    try:
+      val = str(value)
+    except UnicodeEncodeError:
+      try:
+        val = unicode(value)
+      except UnicodeEncodeError:
+        val = strip_accents(value)
```

Comment:
Why not just use utf8 so you know that `val` is always a `str` and never a `unicode`? Upstream processes (read: browsers) should be able to handle utf8 encoded strings.

```
if isinstance(value, basestring):
    val = value.encode('utf8')
else:
    val = str(value)
```